### ISSUE TITLE
nvda controller client under wine always reports nvda available

### ISSUE THREAD

**Issue by lilmike** [nvaccess/nvda#11604]
### Steps to reproduce:

1. Install wine, and run a program that speaks through nvda with the nvda controller client

### Actual behavior:

Nvda is reported as available even when it is not, and, depending on how the program is written, no speech is heard.

### Expected behavior:

Sapi is used as a fallback (if provided by the application) since nvda doesn't work and isn't installed under wine.

#### Did you try to run the COM registry fixing tool in NVDA menu / tools? n/a

More info:

I'm trying to run programs under wine, a windows translator for linux, and if those programs work with nvda for speech output and autodetect which screen reader to use, nvda is always detected as available, even if it is not installed. For
this reason, a sapi fallback, which does work under wine, is never attempted by most programs.

**Comment by lilmike** [comment]
I just looked deeper into the code, and it looks like the testIfRunning method in the library isn't actually implemented, or so it seems. It simply returns 0. Is this intended and/or correct? If not, is there another way to test if nvda is running?

-Michael.

**Comment by leonardder** [comment]
> I just looked deeper into the code, and it looks like the testIfRunning method in the library isn't actually implemented, or so it seems. It simply returns 0. Is this intended and/or correct? If not, is there another way to test if nvda is running?

Yes, this is intended. This is because the controller client uses RPC under the hood. If NVDA is not running and therefore NVDA's RPC server is not, the client library will return an error code, probably something like 1722.

I'd say RPC support would be limited in Wine.

**Comment by lilmike** [comment]
Ok, that makes more sense. As for wine, is there a way to detect if nvda is running that doesn't include the rpc server/protocol? Obviously it would be preferable if wine extended their rpc implementation. I'm thinking something like looking for a background window -- for jaws I think it's something like JFWUI2.
Thanks for the help!
-Michael.

**Comment by feerrenrut** [comment]
> This is because the controller client uses RPC under the hood. If NVDA is not running and therefore NVDA's RPC server is not, the client library will return an error code, probably something like 1722.

This would be a very helpful thing to add to the docs for NVDA Controller. 

>As for wine, is there a way to detect if nvda is running that doesn't include the rpc server/protocol?

In the context of NVDA controller there isn't much point doing that, really this function is "is NVDA running and can we communicate with it."

However, if your goal is to just ensure that NVDA is running, you could just check that the process is running? A more complicated approach is to use a similar approach to the NVDA system tests (an nvda plugin is installed which communicates on a known channel).

Either way, running under wine is not something we would like to try to support right now, so I'm going to close this issue to avoid any confusion for people who might find it. However, I do invite you to continue the conversation here.

**Comment by befabry** [comment]
Hello,

There was a lot of progress these past years regarding Wine / Proton. Is it somehow working now ? As a software engineer working with Linux it would be really interesting as accessibility is way more looked at these past years.

Regards
